src/daft-core/src/array/ops/list.rs

```diff
@@ -669,9 +669,131 @@ impl ListArray {
             None,
         ))
     }
+
+    pub fn list_bool_and(&self) -> DaftResult<BooleanArray> {
+        let child = &self.flat_child;
+        let offsets = self.offsets();
+        let validity = self.validity();
+
+        let mut result = Vec::with_capacity(self.len());
+        let mut result_validity = Vec::with_capacity(self.len());
+
+        for i in 0..self.len() {
+            let is_valid = validity.map_or(true, |v| v.get(i).unwrap());
+            if !is_valid {
+                result.push(false);
+                result_validity.push(false);
+                continue;
+            }
+
+            let start = *offsets.get(i).unwrap() as usize;
+            let end = *offsets.get(i + 1).unwrap() as usize;
+            let slice = child.slice(start, end)?;
+
+            // If slice is empty or all null, return null
+            if slice.is_empty()
+                || slice
+                    .validity()
+                    .map_or(false, |v| v.unset_bits() == slice.len())
+            {
+                result.push(false);
+                result_validity.push(false);
+                continue;
+            }
+
+            // Look for first non-null false value
+            let mut all_true = true;
+            let bool_slice = slice.bool()?;
+            let bool_validity = bool_slice.validity();
+            let bool_data = bool_slice.as_arrow().values();
+            for j in 0..bool_slice.len() {
+                if bool_validity.map_or(true, |v| v.get(j).unwrap()) && !bool_data.get_bit(j) {
+                    all_true = false;
+                    break;
+                }
+            }
+            result.push(all_true);
+            result_validity.push(true);
+        }
+
+        let validity_bitmap = arrow2::bitmap::Bitmap::from_iter(result_validity.iter().copied());
+        let values = arrow2::bitmap::Bitmap::from_iter(result.iter().copied());
+        let arrow_array = arrow2::array::BooleanArray::new(
+            arrow2::datatypes::DataType::Boolean,
+            values,
+            Some(validity_bitmap),
+        );
```

Comment:
This is not related to your PR, but it's extremely strange that you have to give `BooleanArray::new` a datatype... the datatype should be `DataType::Boolean`.

Looking at the implementation, of `BooleanArray::new`, it calls `BooleanArray::try_new` which throws an error if the datatype given is not `DataType::Boolean`.

Reply:
By the way, this comment is not requesting you to change anything; I may create a follow-up PR to clean up the `BooleanArray::new` API in the future.